README.md

```diff
@@ -452,6 +452,166 @@ And the log from the second server shows that it connected to the third.
+
+To illustrate, lets define two clusters named A and B, having one server on each, SA and SB respectively. For this illustration, SA and SB have each an outbound gateway connection to each other, and therefore an inbound gateway connection from each other.
+
+Suppose that a client connects to SB under account `MyAccount` and publishes on subject `foo`. SB sends this message to its client subscriptions (if any) and to its routes if there is an interest in its cluster. Then comes the gateways. For gateway `A`, it checks if there is a registered no-interest on account `MyAccount/foo`. Since this is not the case, it sends the message to SA. When SA receives the message from its inbound gateway connection, it first looks-up the account `MyAccount`. If the account is not registered, SA sends a no-interest protocol for this account so that SB stops sending messages on any subject on this account. Same occurs if the account is registered but there is no subscription to any subject on this account. If there is a subscription interest but say on `bar` but not `foo`, then SA sends a protocol to indicate no interest
+on `foo`. If later a subscription on `foo` is created in cluster A (directly on SA or any server in cluster A), SA knows that it has sent a no-interest protocol on `MyAccount` (or `foo` depending on the situation), so it has to send a protocol to cancel the no-interest.
+
+#### Interest-only mode
+
+Using the example above, if SB sends lot of messages on various subjects for which SA has no interest, SA sends to its inbound gateway connection a protocol message indicating that SB should now stop sending optimistically and instead send only if there is a known interest for the subject on SA. SA sends its list of plain subscriptions and will now do so anytime there is a new subscription, or when the last subscription on a given subject is gone.
+
+#### Queue subscriptions
+
+Queue subscriptions behave as if the inbound gateway connection was in interest-only mode, that is, any local or cluster-wide queue subscription is sent to a server's inbound gateway connections. With the example above, if a client connected to SA creates a queue subscription on subject `foo` with queue name `bar`, a subscription interest is sent to SB. Note that there is a single protocol per account, per subject and per queue name. When the last of the queue subscription is gone, an unsubscribe protocol is sent to SB.
+
+Why queue differs from plain subscriptions? Because queue subscriptions interest is global and still honor the fact that a given message must be delivered to one and only one queue subscriber on a given subject for the same queue name.
+
+However, unlike in a simple cluster where a server randomly sends a message to a given queue group based on the total number of queue subscribers, with gateways, if a queue group exists in several clusters, the message is delivered only to the local cluster
```

Comment:
Message delivery is preferred to local queue subscribers to which the message producer is connected.